packages/client/src/transaction/helpers/index.ts

```diff
@@ -0,0 +1,47 @@
+import { RelayerResultFragment, RelayErrorFragment } from '../../graphql/fragments.generated';
+import {
+  TransactionErrorFragment,
+  TransactionIndexedResultFragment,
+} from '../graphql/transaction.generated';
+
+export function isRelayerResult(
+  result: RelayerResultFragment | RelayErrorFragment,
+): result is RelayerResultFragment {
+  return 'txId' in result;
+}
+
+export function isRelayerError(
+  result: RelayerResultFragment | RelayErrorFragment,
+): result is RelayErrorFragment {
+  return 'reason' in result;
+}
+
+export function isTransactionIndexedResult(
+  result: TransactionIndexedResultFragment | TransactionErrorFragment,
+): result is TransactionIndexedResultFragment {
+  return 'indexed' in result;
+}
+
+export function isTransactionError(
+  result: TransactionIndexedResultFragment | TransactionErrorFragment,
+): result is TransactionErrorFragment {
+  return 'reason' in result;
+}
+
+export function getTxIdFromRelayerResult(
+  result: RelayerResultFragment | RelayErrorFragment,
+): string {
+  if (isRelayerResult(result)) {
+    return result.txId;
+  }
+  throw new Error(result.reason);
+}
+
+export function getIsIndexedFromTransactionResult(
```

Comment:
We have the tendency of naming helpers that take a lump of data and either extract part of it or figure out an a aspect of it (e.g. resolves a discriminated union) with the prefix "get".

These are not technically getters, I think we should be more precise in general: `resolve`, `extract` are better suited in most of the cases in this PR.